File: .vscode/functions/financial-calc_update.c
//andrea lugo, financial_calc_update.c
#include <stdio.h>

float use(char type[]) {
    float amount;
    printf("What is your %s amount?\n", type);
    scanf("%f", &amount);
    return amount;
}

void word(float income, float amount, char type[]) {
    float percent = (amount / income) * 100;
    printf("You are spending $%.2f on %s, which is %.2f%% of your income.\n", amount, type, percent);
}

int main(void) {
    printf("Welcome to the Financial Calculator!\n");

    float income = use("income");
    float rent = use("rent");
    float utilities = use("utilities");
    float groceries = use("groceries");
    float transportation = use("transportation");

    float savings = income * 0.10;

    float spending = rent + utilities + groceries + transportation;

    word(income, rent, "rent");
    word(income, utilities, "utilities");
    word(income, groceries, "groceries");
    word(income, transportation, "transportation");

    printf("You are saving $%.2f, which is 10%% of your income.\n", savings);
    printf("Your total spending is $%.2f, excluding savings.\n", spending);

    return 0;
}

File: .vscode/functions/financial_calc_update.py
# Andrea lugo, financial calculator python program

def get_user_inputs():
    
    try:
        income = float(input("Enter your monthly income: $"))
        expenses = {}
        print("Enter your expenses. Type 'done' when finished.")
        
        while True:
            category = input("Expense category: ").strip().lower()
            if category == 'done':
                break
            try:
                amount = float(input(f"Amount for {category}: $"))
                expenses[category] = amount
            except ValueError:
                print("Invalid input. Please enter a numeric value.")
        
        return income, expenses
    except ValueError:
        print("Invalid input. Please enter a numeric value.")
        return get_user_inputs()

def calculate_percentages(income, expenses):
   
    print("\n--- Budget Breakdown ---")
    for category, amount in expenses.items():
        percentage = (amount / income) * 100 if income > 0 else 0
        print(f"{category.capitalize()}: ${amount:.2f} ({percentage:.0f}%)")

    total_expenses = sum(expenses.values())
    savings = income - total_expenses
    savings_percentage = (savings / income) * 100 if income > 0 else 0
    print(f"\nTotal Expenses: ${total_expenses:.2f}")
    print(f"Savings: ${savings:.2f} ({savings_percentage:.0f}%)\n")

def main():
    """
    Main function to run the budget calculator.
    """
    print("Welcome to the Budget Calculator!")
    income, expenses = get_user_inputs()
    calculate_percentages(income, expenses)

# Run the program
if __name__ == "__main__":
    main()